postgres/parser/uuid/codec.go

```diff
@@ -122,6 +122,17 @@ func (u *UUID) UnmarshalText(text []byte) error {
 	case 41, 45:
 		return u.decodeURN(text)
 	default:
+		shortenedText := bytes.ReplaceAll(text, []byte{'-'}, nil)
```

Comment:
Why is this necessary? Does the parser not actually support the forms it claims to in the method doc?

Reply:
If this is to support another form such as `f47ac10b58cc4372a567-0e02b2c3d479`, then you should make that explicit. If postgres is more lenient about what it will parse for a UUID that's fine, but in that case you should generalize your block here to the entire method, not the default case